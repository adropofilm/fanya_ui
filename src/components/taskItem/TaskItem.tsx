import styles from "./TaskItem.module.css";
import { Dispatch, ReactElement, SetStateAction } from "react";
import { Task, Status } from "../../types/Task.types";
import { deleteTask, updateTask } from "../../utility/api";

export type Props = Task & {
  setTasks: Dispatch<SetStateAction<Task[]>>;
};

export const TaskItem = ({ id, title, setTasks }: Props): ReactElement => {
  const markTaskCompleted = async (): Promise<void> => {
    const taskBody = {
      status: Status.CLOSED,
      id,
    };

    try {
      await updateTask(taskBody);
      setTasks((prevState) => prevState.filter((task) => task.id !== id));
    } catch (error) {
      console.error(error);
    }
  };

  const markTaskDeleted = async (): Promise<void> => {
    try {
      await deleteTask(id as number);
      setTasks((prevState) => prevState.filter((task) => task.id !== id));
    } catch (error) {
      console.error(error);
    }
  };

  return (
    <div className={styles["container"]}>
      <input className="checkbox-input" role="checkbox" aria-checked="false" />
      <span className={styles["checkmark"]} onClick={markTaskCompleted}></span>
      <label>{title}</label>
      <span
        className={`material-symbols-outlined ${styles["delete-icon"]}`}
        onClick={markTaskDeleted}>
        delete
      </span>
    </div>
  );
};
