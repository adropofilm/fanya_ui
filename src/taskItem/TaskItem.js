import React from "react";
import './styles.css';

const TaskItem = ({id, status, title}) => {

    return (
        <div key={id} className="task-item">
            <p>{title}</p>
        </div>
    )
}

export default TaskItem;