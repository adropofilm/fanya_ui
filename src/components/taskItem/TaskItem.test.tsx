import renderer from "react-test-renderer";
import { TaskItem } from "./TaskItem";

const setTasks = jest.fn();

it("renders correctly", () => {
  const tree = renderer
    .create(
      <TaskItem
        id={1}
        status="open"
        title="clean the garage"
        setTasks={setTasks}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
