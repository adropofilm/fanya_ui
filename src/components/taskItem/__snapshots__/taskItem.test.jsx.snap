// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <input
    aria-checked="false"
    className="checkbox-input"
    role="checkbox"
  />
  <span
    className="checkmark"
    onClick={[Function]}
  />
  <label>
    clean the garage
  </label>
  <span
    className="material-symbols-outlined delete-icon"
    onClick={[Function]}
  >
    delete
  </span>
</div>
`;
